src/hash/zelbet_gadget/bar.rs

```diff
@@ -69,21 +67,24 @@ fn witness_decomposition_gadget(
 
 /// In circuit bar function, making use of decomposition gadget that is defined
 /// in PLONK
-pub fn bar_gadget(
-    composer: &mut TurboComposer,
-    state: &mut [Witness; 3],
-    zelbet_witnesses: &ZelbetWitnesses,
-) {
+pub fn bar_gadget(composer: &mut TurboComposer, state: &mut [Witness; 3]) {
+    // Array to hold witnesses for 0, 1 and 2 so we don't need to append them
+    // multiple times
+    let mut witnesses = [TurboComposer::constant_zero(); 3];
+    witnesses[1] = composer.append_witness(BlsScalar::one());
+    witnesses[2] = composer.append_witness(BlsScalar::from(2));
```

Comment:
These are constants and shouldn't be witnesses - otherwise a malicious prover might be able to tamper the proof.
```suggestion
    witnesses[1] = composer.append_constant(BlsScalar::one());
    witnesses[2] = composer.append_constant(BlsScalar::from(2));
```